src/glview.cpp

```diff
@@ -943,6 +1015,71 @@ void GLView::zoom( float z )
 	update();
 }
 
+void GLView::setVertexPaintSettings(GLView::PaintSettings settings)
+{
+    cfg.vertexPaintSettings = settings;
+    update();
+}
+
+void GLView::startVertexPaint(const QPoint& point)
+{
+    mousePaint = true;
+    mousePaintVerts.clear();
+    mousePaintHitDetectImg = renderIndexImage();
+    vertexPaint(point);
+    QCursor cursor(Qt::BlankCursor);
+    QApplication::setOverrideCursor(cursor);
+    QApplication::changeOverrideCursor(cursor);
+}
+
+void GLView::vertexPaint(const QPoint& point)
+{
+    bool updated = false;
+    if (mousePaint)
+    {
+        // Let's do an absolutely shit job and act like it's decent
+        auto inds = sampleIndexImageCircle(mousePaintHitDetectImg, point, cfg.vertexPaintSettings.brushSize, model);
+
+        for (const auto& ind : inds)
+        {
+            if ( ind.isValid() && !mousePaintVerts.contains(ind))
+            {
+                mousePaintVerts.push_back(ind);
+                Color4 base = model->get<ByteColor4>(ind.parent(), "Vertex Colors");
+                Color4 blend;
+
+                if (cfg.vertexPaintSettings.brushMode == PaintBlendMode::BlendNormal)
+                {
+                   blend = (cfg.vertexPaintSettings.brushColor * cfg.vertexPaintSettings.brushOpacity) + (base * (Color4() - cfg.vertexPaintSettings.brushOpacity));
```

Comment:
(cfg.vertexPaintSettings.brushColor * cfg.vertexPaintSettings.brushOpacity) can be pulled outside of the for loop.
Could also assign cfg.vertexPaintSettings.brushOpacity to a variable as used in a few locations aswell.